### PR TITLE
[Notifier] Add SMS options to AllMySms notifier

### PR DESCRIPTION
**Pull request by gnito-org**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | 
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+final class AllMySmsOptions implements MessageOptionsInterface
+{
+    private array $options;
+
+    public function __construct(array $options = [])
+    {
+        $this->options = $options;
+    }
+
+    public function getAlerting(): ?int
```
Comment (qdequippe) [inline src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:28]: are getters needed?

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        $this->options['verbose'] = $verbose;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        $options = $this->options;
+        if (isset($options['recipient_id'])) {
+            unset($options['recipient_id']);
```
Comment (qdequippe) [inline src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:140]: why this unset?

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsTransport.php [hunk 3]
```diff
@@ -63,14 +63,31 @@ protected function doSend(MessageInterface $message): SentMessage
...
         $from = $message->getFrom() ?: $this->from;
 
+        $opts = $message->getOptions();
+        $options = $opts ? $opts->toArray() : [];
+        $options['from'] = $options['from'] ?? $from;
+        $options['to'] = $message->getPhone();
+        $options['text'] = $message->getSubject();
+
+        if (isset($options['campaign_name'])) {
+            $options['campaignName'] = $options['campaign_name'];
```
Comment (qdequippe) [inline src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsTransport.php:73]: why not use `campaignName` key directly in AllMySmsOptions?

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php (hunk 2 above)
Comment (gnito-org) [reply src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:140]: `recipient_id` is never used to override `$message->getPhone()`.

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php (hunk 1 above)
Comment (gnito-org) [reply src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:28]: Just following the precedent that is set in `MessageInterface`.

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsTransport.php (hunk 3 above)
Comment (gnito-org) [reply src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsTransport.php:73]: To maintain a snake-case standard for all array indexes in the options classes.

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php (hunk 2 above)
Comment (qdequippe) [reply src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:140]: So why the setter is needed in this class?

```
public function setRecipientId(string $id): self
{
     $this->options['recipient_id'] = $id;

     return $this;
}
```

File: src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php (hunk 1 above)
Comment (gnito-org) [reply src/Symfony/Component/Notifier/Bridge/AllMySms/AllMySmsOptions.php:28]: I meant MessageOptionsInterface.

**Comment by nicolas-grekas** [comment]
Thank you @gnito-org.

**Pull request merged**